File: packages/core/tests/fixtures/unhealthy/exception-antipatterns.ts
// Sprint 58 fixture: exception anti-patterns in TypeScript
// Expected: >= 2 ExceptionHandlingAntiPattern smells

// 1. EmptyCatch — catch block with empty body
function parseData(raw: string): unknown {
  try {
    return JSON.parse(raw);
  } catch (e) {}
}

// 2. EmptyCatch with comment only
function loadConfig(path: string): Record<string, unknown> {
  try {
    const fs = require('fs');
    return JSON.parse(fs.readFileSync(path, 'utf8'));
  } catch (err) {
    // TODO: handle error later
  }
  return {};
}

// 3. DestructiveWrapping — re-throw loses stack trace
function fetchUser(id: string): Promise<unknown> {
  try {
    return fetch(`/api/users/${id}`);
  } catch (e) {
    throw new Error(e.message);
  }
}

// 4. Generic catch — catches all errors without discrimination
function processRequest(req: { body: unknown }): void {
  try {
    handleBody(req.body);
  } catch (error) {
    console.log('Something went wrong');
  }
}

function handleBody(_body: unknown): void {
  // implementation
}

File: packages/core/tests/fixtures/unhealthy/exception-antipatterns.py
# Sprint 58 fixture: exception anti-patterns in Python
# Expected: >= 1 ExceptionHandlingAntiPattern smell

import json


def parse_config(path: str) -> dict:
    """Load configuration from file."""
    try:
        with open(path) as f:
            return json.load(f)
    except:
        pass  # EmptyCatch with bare except + pass


def load_user(user_id: int) -> dict:
    """Fetch user from database."""
    try:
        return db_query(f"SELECT * FROM users WHERE id = {user_id}")
    except Exception as e:
        pass  # CatchGeneric — catches Exception with pass


def transform_data(raw: str) -> dict:
    """Transform raw input to structured data."""
    try:
        return json.loads(raw)
    except json.JSONDecodeError as e:
        raise RuntimeError(str(e))  # DestructiveWrapping — no "from e"


def db_query(sql: str) -> dict:
    """Stub database query."""
    return {}
